esmvaltool/diag_scripts/primavera/eady_growth_rate/eady_growth_rate.py

```diff
@@ -0,0 +1,210 @@
+        # along pressure levels
+
+        plevs = x.shape[1]
+
+        dxdy_0 = ((x[:, 1, :, :].lazy_data() - x[:, 0, :, :].lazy_data()) /
+                  (y[:, 1, :, :].lazy_data() - y[:, 0, :, :].lazy_data()))
+
+        dxdy_centre = ((x[:, 2:plevs, :, :].lazy_data() -
+                        x[:, 0:plevs - 2, :, :].lazy_data()) /
+                       (y[:, 2:plevs, :, :].lazy_data() -
+                        y[:, 0:plevs - 2, :, :].lazy_data()))
+
+        dxdy_end = ((x[:, plevs - 1, :, :].lazy_data() -
+                     x[:, plevs - 2, :, :].lazy_data()) /
+                    (y[:, plevs - 1, :, :].lazy_data() -
+                     y[:, plevs - 2, :, :].lazy_data()))
+
+        bounds = [dxdy_end, dxdy_0]
+        stacked_bounds = da.stack(bounds, axis=1)
+        total = [dxdy_centre, stacked_bounds]
+
+        # Concatenate arrays where the last slice is dxdy_0
+        dxdy = da.concatenate(total, axis=1)
+
+        # Move dxdy_0 to the beggining of the array
+        dxdy = da.roll(dxdy, 1, axis=1)
+
+        return dxdy
+
+    def brunt_vaisala_frq(self, theta, zg):
+        dthdz = self.vertical_integration(theta, zg)
+        dthdz = da.where(dthdz > 0, dthdz, 0)
+        buoy = (self.g / theta.lazy_data()) * dthdz
+        brunt = da.sqrt(buoy)
+        brunt = da.where(brunt != 0, brunt, self.fill_value)
+
+        return brunt
+
+    def coriolis(self, lats, ndim):
+        fcor = 2.0 * self.omega * np.sin(np.radians(lats.points))
+        fcor = fcor[np.newaxis, np.newaxis, :, np.newaxis]
+        fcor = da.broadcast_to(fcor, ndim)
+
+        return fcor
+
+    def eady_growth_rate(self, fcor, ua, zg, brunt):
+        dudz = self.vertical_integration(ua, zg)
+        egr = self.con * abs(fcor) * abs(dudz) / brunt
+
+        return egr
+
+    def seasonal_plots(self, egr, alias):
+        try:
+            levels = self.cfg['plot_levels']
+        except KeyError:
+            logger.info("Parameter plot_levels is not set in the recipe."
+                        "Plotting all pressure levels instead.")
+            levels = egr.coord('air_pressure').points
+        for level in levels:
+            cube = extract_levels(egr, level, scheme='linear')
+            crs_latlon = ccrs.PlateCarree()
+            ax = plt.axes(projection=ccrs.PlateCarree())
+            ax.coastlines(linewidth=1, color='black')
+            # North Atlantic
+            ax.set_extent((-90.0, 30.0, 20.0, 80.0), crs=crs_latlon)
+            ax.set_yticks(np.linspace(25, 75, 6))
+            qplt.contourf(cube, levels=np.arange(0, np.max(cube.data), 0.05))
+            extension = self.cfg['output_file_type']
+            diagnostic = self.cfg['script']
+            plotname = '_'.join([alias, diagnostic,
+                                 str(int(level))]) + f'.{extension}'
+            plt.savefig(os.path.join(self.cfg[n.PLOT_DIR], plotname))
+            plt.close()
+
+    def save(self, egr, alias, data):
```

Comment:
This one is fairly obvious, but you could add a short docstring on what it expects and does.